app/scripts/modules/core/src/instance/instance.write.service.ts

```diff
@@ -69,7 +69,7 @@ export class InstanceWriter {
     return this.executeMultiInstanceTask(instanceGroups, application, 'rebootInstances', 'Reboot');
   }
 
-  public rebootInstance(instance: IInstance, application: Application, params: any = {}) {
+  public rebootInstance(instance: IInstance, application: Application, params: any = {}): ng.IPromise<ITask> {
```

Comment:
add `IPromise` to the angular import line at the top, and don't do `ng.`.

Reply:
should i change all the other functions too?

Reply:
Yes please